warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentOrderedEvaluatorTest.java

```diff
@@ -0,0 +1,262 @@
+        List<TermWeightPosition> offsetsA = asList(1);
+        List<TermWeightPosition> offsetsB = asList(2);
+        List<TermWeightPosition> offsetsC = asList(3);
+        
+        offsets.add(offsetsA);
+        offsets.add(offsetsB);
+        offsets.add(offsetsC);
+        
+        evaluator = new WrappedContentOrderedEvaluator(null, 3, new HashMap<>(), "a", "b", "c");
+        
+        Assert.assertTrue(evaluator.evaluate(offsets));
+    }
+    
+    @Test
+    public void evaluate_simpleSuccessDistance3FailTest() {
+        List<List<TermWeightPosition>> offsets = new ArrayList<>();
+        List<TermWeightPosition> offsetsA = asList(1);
+        List<TermWeightPosition> offsetsB = asList(5);
+        List<TermWeightPosition> offsetsC = asList(7);
+        
+        offsets.add(offsetsA);
+        offsets.add(offsetsB);
+        offsets.add(offsetsC);
+        
+        evaluator = new WrappedContentOrderedEvaluator(null, 3, new HashMap<>(), "a", "b", "c");
+        
+        Assert.assertTrue(!evaluator.evaluate(offsets));
+    }
+    
+    @Test
+    public void evaluate_pruneTopTest() {
+        List<List<TermWeightPosition>> offsets = new ArrayList<>();
+        List<TermWeightPosition> offsetsA = asList(1, 10);
+        List<TermWeightPosition> offsetsB = asList(10, 11);
+        List<TermWeightPosition> offsetsC = asList(4, 12);
+        
+        offsets.add(offsetsA);
+        offsets.add(offsetsB);
+        offsets.add(offsetsC);
+        
+        evaluator = new WrappedContentOrderedEvaluator(null, 1, new HashMap<>(), "a", "b", "c");
+        
+        Assert.assertTrue(evaluator.evaluate(offsets));
+    }
+    
+    @Test
+    public void evaluate_pruneBottomTest() {
+        List<List<TermWeightPosition>> offsets = new ArrayList<>();
+        List<TermWeightPosition> offsetsA = asList(1, 3, 8);
+        List<TermWeightPosition> offsetsB = asList(3, 4, 44);
+        List<TermWeightPosition> offsetsC = asList(4, 5, 35);
+        
+        offsets.add(offsetsA);
+        offsets.add(offsetsB);
+        offsets.add(offsetsC);
+        
+        evaluator = new WrappedContentOrderedEvaluator(null, 1, new HashMap<>(), "a", "b", "c");
+        
+        Assert.assertTrue(evaluator.evaluate(offsets));
+    }
+    
+    @Test
+    public void evaluate_pruneTopAndBottomTest() {
+        List<List<TermWeightPosition>> offsets = new ArrayList<>();
+        List<TermWeightPosition> offsetsA = asList(3, 4, 5, 6);
+        List<TermWeightPosition> offsetsB = asList(7, 9, 22);
+        List<TermWeightPosition> offsetsC = asList(8, 11, 13);
+        
+        offsets.add(offsetsA);
+        offsets.add(offsetsB);
+        offsets.add(offsetsC);
+        
+        evaluator = new WrappedContentOrderedEvaluator(null, 1, new HashMap<>(), "a", "b", "c");
+        
+        Assert.assertTrue(evaluator.evaluate(offsets));
+    }
+    
+    @Test
+    public void evaluate_livePruneTest() {
```

Comment:
Can we add a test where we need to alternate the match please
A = 1, 10, 20
B = 21, 24, 30
C = 3, 8, 22

Reply:
Added 2 new tests for this, for FORWARD and REVERSE variants